### DISCUSSION TITLE
Advantages / Disadvantages of adding google closure compiler

### DISCUSSION THREAD

**Discussion by xzilja** [vercel/next.js#10717] [Help] [answered]
I've been reading up about [Google's closure compiler](https://developers.google.com/closure/compiler) and to my understanding it can produce more performant and smaller javascript bundles.

It looks like it is relatively easy to add alongside next js via https://github.com/webpack-contrib/closure-webpack-plugin

However there will probably be multiple places where babel config needs to be adjusted, hence I wanted to ask if it makes sense to try this out, would it improve performance / bundle size at all or is next js already doing better job here?

**Comment by timneutkens** [comment, marked as answer]
It won't hold *massive* improvements and will likely increase your build times significantly if applied in this way. In general I wouldn't recommend changing the webpack configuration. We're going to introduce more improvements soon.

Basically if you want to optimize your application it'd be more productive to get rid of excessive node_modules and optimizing application code.

**Comment by ilangleben19** [comment]
Sorry to revive a dead-ish thread. How theoretically would one go about using the Closure Compiler if compile times weren't an issue?
